pycc/rt/rtcc.py

```diff
@@ -126,42 +127,49 @@ def f(self, t, y):
         if self.ccwfn.local is not None:
             rl1, rl2 = self.ccwfn.Local.filter_res(rl1, rl2)
 
+        # Phase contribution = exp(-phase(t))
+        phase = self.phase(F, t1, t2)
+
         # Pack up the residuals
-        y = self.collect_amps(rt1, rt2, rl1, rl2)
+        y = self.collect_amps(rt1, rt2, rl1, rl2, phase)
 
         return y
 
-    def collect_amps(self, t1, t2, l1, l2):
+    def collect_amps(self, t1, t2, l1, l2, phase):
         """
         Parameters
         ----------
+        phase : scalar
+            current wave function phase
         t1, t2, l2, l2 : NumPy arrays
             current cluster amplitudes or residuals
 
         Returns
         -------
         NumPy array
-            amplitudes or residuals as a vector (flattened array)
+            amplitudes or residuals and phase as a vector (flattened array)
         """
         if isinstance(t1, torch.Tensor):
             t1 = torch.flatten(t1)
             t2 = torch.flatten(t2)
             l1 = torch.flatten(l1)
             l2 = torch.flatten(l2)
-            return torch.cat((t1, t2, l1, l2)).type(torch.complex128)
+            return torch.cat((t1, t2, l1, l2, phase)).type(torch.complex128)
         else:
-            return np.concatenate((t1, t2, l1, l2), axis=None).astype('complex128')
+            return np.concatenate((t1, t2, l1, l2, phase), axis=None).astype('complex128')
 
 
     def extract_amps(self, y):
         """
         Parameters
         ----------
         y : NumPy array
-            flattened array of cluster amplitudes or residuals
+            flattened array of wave function phase and cluster amplitudes or residuals
```

Comment:
`amplitudes or residuals and phase as a vector (flattened array)` 1) for consistency with `collect_amps`, and 2) to imply `phase` is the last element in the vector